maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java

```diff
@@ -150,6 +150,7 @@ public DefaultRepositorySystemSession newRepositorySession( MavenExecutionReques
         configProps.put( ConfigurationProperties.USER_AGENT, getUserAgent() );
         configProps.put( ConfigurationProperties.INTERACTIVE, request.isInteractiveMode() );
         configProps.put( "maven.startTime", request.getStartTime() );
+        // TODO Why is this duplication necessary?
```

Comment:
Well, yes, but it will not work. I have to pass both properties to config props since Resolver uses `ConfigUtils` which uses `session.getConfigProperties()` only. `get(User|System)Properties()` are never invoked from the `session`. I will change the `TODO` to a decent comment.

Reply:
Added comment.